www/includes/easyparliament/member.php

```diff
@@ -242,6 +242,17 @@ public function date_in_memberships($house, $date){
         return false;
     }
 
+    public function member_agreements($policyID, $house = HOUSE_TYPE_COMMONS){
```

Comment:
Could we pass in the Policies object rather than only a policy ID? Only each time it's instantiated it has to load in the big JSON file, so eg the call from Divisions.php is doing that over and over once per policy ID, but you've already created such an object on new line 261 (and the other call, you already have $policiesList).

Reply:
Thinking about this - is the core problem that it's become expensive to create that object? Would putting json-based data in a 5 minute memcache sidestep this? (appreciate that's adding complexity to an already unideal approach).

Otherwise yes, this method could accept an optional policies object and only create a new one if it didn't get it.

Reply:
I don't know if it's actually expensive, it just felt wrong to repeatedly open a file. And sticking in a cache also feels wrong for the same reason :) Feels like you'd only ever need to instantiate the policies once each request, but meh :) opitonal policies object sounds good.

Reply:
Null approach added!